### PR TITLE
Persisting created new campaignApplication to db

### PR DESCRIPTION
**Pull request by Martbul**

### CODE REVIEW

File: apps/api/src/campaign-application/campaign-application.controller.spec.ts [hunk 1]
```diff
@@ -26,63 +42,61 @@ describe('CampaignApplicationController', () => {
     expect(controller).toBeDefined()
   })
 
-  it('when create called it should delegate to the service create', () => {
```
Comment (gparlakov) [inline apps/api/src/campaign-application/campaign-application.controller.spec.ts:29]: This is deleted but the logic is still in use. Was this by design?

File: apps/api/src/campaign-application/campaign-application.controller.spec.ts [hunk 2]
```diff
@@ -2,21 +2,37 @@ import { Test, TestingModule } from '@nestjs/testing'
 import { CampaignApplicationController } from './campaign-application.controller'
 import { CampaignApplicationService } from './campaign-application.service'
 import { SpyOf, autoSpy } from '@podkrepi-bg/testing'
```
Comment (gparlakov) [inline apps/api/src/campaign-application/campaign-application.controller.spec.ts:4]: Some unuses stuff that github(or an action?) points out - SpyOf and CamapignTypeCategroy

File: apps/api/src/campaign-application/campaign-application.controller.spec.ts [hunk 3]
```diff
@@ -26,63 +42,61 @@ describe('CampaignApplicationController', () => {
...
   })
 
   it('when findOne called it should delegate to the service findOne', () => {
-    // arrange
-    // act
+    // Act
     controller.findOne('id')
 
-    // assert
+    // Assert
```
Comment (gparlakov) [inline apps/api/src/campaign-application/campaign-application.controller.spec.ts:91]: avoid such unnecessary changes - they make the code review that much harder

File: apps/api/src/campaign-application/campaign-application.service.spec.ts [hunk 4]
```diff
@@ -20,131 +41,117 @@ describe('CampaignApplicationService', () => {
...
+
+      expect(result).toEqual(mockCreatedCampaignApplication)
+
+      expect(mockOrganizerService.create).toHaveBeenCalledWith({
+        personId: mockPerson.id,
+      })
+
+      expect(prismaMock.campaignApplication.create).toHaveBeenCalledWith({
+        data: {
+          campaignName: dto.campaignName.trim(),
```
Comment (gparlakov) [inline apps/api/src/campaign-application/campaign-application.service.spec.ts:130]: This looks like the logic itself and while it's technically correct I belive we'll get more value having hardcoded value here i.e.  {

data :{
campaignName: 'my-campaign',
organizerName: 'test organizer',

...

File: apps/api/src/campaign-application/campaign-application.service.ts [hunk 5]
```diff
@@ -1,24 +1,64 @@
...
+      })
+
+      if (!organizer) {
+        organizer = await this.organizerService.create({
+          personId: person.id,
+        })
+      }
+
+      const sanitizedData = {
+        campaignName: createCampaignApplicationDto.campaignName.trim(),
```
Comment (gparlakov) [inline apps/api/src/campaign-application/campaign-application.service.ts:36]: @sashko9807 is there a sanitizing logic somewhere on the stack (nest, middleware, prisma perhaps)  that prevents Cross Site Scripting or SQL injection (i.e. my name is <script>alert('pawned')</script> or the famous [jonny drop tables](https://stackoverflow.com/questions/332365/how-does-the-sql-injection-from-the-bobby-tables-xkcd-comic-work)

Comment (sashko9807) [reply apps/api/src/campaign-application/campaign-application.service.ts:36]: Prisma sanitizes data on insertion. Although I am not sure whether the start-end spaces are automatically trimmed.  Don't recall seeing this practice in any other part of the codebase

Will have to check for that.

File: apps/api/src/campaign-application/campaign-application.service.ts [hunk 6]
```diff
@@ -1,24 +1,64 @@
...
+        where: { personId: person.id },
+      })
+
+      if (!organizer) {
+        organizer = await this.organizerService.create({
+          personId: person.id,
+        })
+      }
+
+      const sanitizedData = {
```
Comment (sashko9807) [inline apps/api/src/campaign-application/campaign-application.service.ts:35]: Prisma sanitizes the input by default. Seems like it also trims unnecessary spaces from user input, thus `sanitizedData` is not needed.

**Review by sashko9807 — CHANGES_REQUESTED** [review]

**Review by sashko9807 — CHANGES_REQUESTED** [review]

**Review by sashko9807 — APPROVED** [review]
